azure-pipelines.yml

```diff
@@ -13,38 +13,54 @@ strategy:
   matrix:
     linux_gcc6:
       VM_ImageName: 'ubuntu-16.04'
-      Compiler_ImageName: 'axom/compilers:gcc-6'
-      CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
+      Compiler_ImageName: 'axom/tpls:gcc-6'
+      CMAKE_EXTRA_FLAGS: ''
       COMPILER: 'g++'
       TEST_TARGET: 'linux_gcc6'
-    linux_gcc7:
-      VM_ImageName: 'ubuntu-16.04'
-      Compiler_ImageName: 'axom/compilers:gcc-7'
-      CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
-      COMPILER: 'g++'
-      TEST_TARGET: 'linux_gcc7'
+      HOST_CONFIG: 'docker-linux-ubuntu16.04-x86_64-gcc@6.1.0'
     linux_gcc8:
       VM_ImageName: 'ubuntu-16.04'
-      Compiler_ImageName: 'axom/compilers:gcc-8'
-      CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
+      Compiler_ImageName: 'axom/tpls:gcc-8'
+      CMAKE_EXTRA_FLAGS: ''
       COMPILER: 'g++'
       TEST_TARGET: 'linux_gcc8'
-    linux_clang4:
+      HOST_CONFIG: 'docker-linux-ubuntu16.04-x86_64-gcc@8.1.0'
+    linux_gcc7:
       VM_ImageName: 'ubuntu-16.04'
-      Compiler_ImageName: 'axom/compilers:clang-4'
-      CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
-      COMPILER: 'clang++'
-      TEST_TARGET: 'linux_clang4'
+      Compiler_ImageName: 'axom/tpls:gcc-7'
+      CMAKE_EXTRA_FLAGS: ''
+      COMPILER: 'g++'
+      TEST_TARGET: 'linux_gcc7'
+      HOST_CONFIG: 'docker-linux-ubuntu16.04-x86_64-gcc@7.3.0'
+    linux_clang4:
+       VM_ImageName: 'ubuntu-16.04'
+       Compiler_ImageName: 'axom/tpls:clang-4'
+       CMAKE_EXTRA_FLAGS: ''
+       COMPILER: 'clang++'
+       TEST_TARGET: 'linux_clang4'
+       HOST_CONFIG: 'docker-linux-ubuntu16.04-x86_64-clang@4.0.0'
+    linux_clang5:
+       VM_ImageName: 'ubuntu-16.04'
+       Compiler_ImageName: 'axom/tpls:clang-5'
+       CMAKE_EXTRA_FLAGS: ''
+       COMPILER: 'clang++'
+       TEST_TARGET: 'linux_clang5'
+       HOST_CONFIG: 'docker-linux-ubuntu16.04-x86_64-clang@5.0.0'
     linux_clang6:
-      VM_ImageName: 'ubuntu-16.04'
-      Compiler_ImageName: 'axom/compilers:clang-6'
-      CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
-      COMPILER: 'clang++'
-      TEST_TARGET: 'linux_clang6'
+       VM_ImageName: 'ubuntu-16.04'
+       Compiler_ImageName: 'axom/tpls:clang-6'
+       CMAKE_EXTRA_FLAGS: ''
+       COMPILER: 'clang++'
+       TEST_TARGET: 'linux_clang6'
+       HOST_CONFIG: 'docker-linux-ubuntu16.04-x86_64-clang@6.0.0'
     osx_gcc:
       VM_ImageName: 'macos-10.13'
       CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
       TEST_TARGET: 'osx_gcc'
+    windows:
+      VM_ImageName: 'windows-2019'
+      CMAKE_EXTRA_FLAGS: '-DAXOM_ENABLE_SIDRE=Off'
```

Comment:
Is there a particular reason to turn off Sidre on OS X and windows?

Reply:
@gzagaris -- I think the issue is that we're not building TPLs on these platforms and ``sidre`` requires ``conduit``.  

We'll have to fix this in a separate PR.  
* @agcapps is working on the Windows part.
* For OS X -- does anyone know if docker/azure supports this?